backoffconnector_test.go

```diff
@@ -37,11 +37,19 @@ func (h *maxDialHost) Connect(ctx context.Context, ai peer.AddrInfo) error {
 	return h.Host.Connect(ctx, ai)
 }
 
-func getNetHosts(t *testing.T, ctx context.Context, n int) []host.Host {
+type hosts []host.Host
```

Comment:
You don't need this, if you add this to `genNetHosts` (and remove the `defer` statements then):
```go
t.Cleanup(func() { h.Close() })